pkg/services/oauthtoken/oauth_token.go

```diff
@@ -212,8 +216,12 @@ func (o *Service) tryGetOrRefreshAccessToken(ctx context.Context, usr *login.Use
 
 	persistedToken := buildOAuthTokenFromAuthInfo(usr)
 
+	start := time.Now()
 	// TokenSource handles refreshing the token if it has expired
 	token, err := connect.TokenSource(ctx, persistedToken).Token()
+	duration := time.Since(start)
+	o.tokenRefreshDuration.WithLabelValues(authProvider, fmt.Sprintf("%t", err == nil)).Observe(duration.Seconds())
```

Comment:
Do we only want to capture the refresh duration for the metric or would it make more sense to capture the duration of the whole function?

Reply:
Yes, with this I'd to capture the roundtrip time of the outgoing request.